client/signup/jetpack-connect/authorize-form.jsx

```diff
@@ -478,10 +478,10 @@ const LoggedInForm = React.createClass( {
 	renderStateAction() {
 		const { authorizeSuccess, siteReceived } = this.props.jetpackConnectAuthorize;
 		if (
-			this.props.isFetchingSites ||
+			this.props.isFetchingSite ||
 			this.isAuthorizing() ||
 			this.retryingAuth ||
-			( authorizeSuccess && ! siteReceived )
+			( this.props.isFetchingSites && authorizeSuccess && ! siteReceived )
```

Comment:
These conditions seem confusing to me. After thinking about it for a long time, I think what they mean is:

* `this.props.isFetchingSite`: We're trying to fetch the requested auth site
* `( this.props.isFetchingSites && authorizeSuccess && ! siteReceived )`: The JP connect request has been authorized, but we haven't received the site data yet (from where?) and we're still waiting on global sites state to be updated.

If that third condition is true, we'll wind up with one of these messages, is that correct and the desired behavior?

```js
if ( authorizeSuccess && isRedirectingToWpAdmin ) {
	return this.translate( 'Returning to your site' );
}

if ( authorizeSuccess ) {
	return this.translate( 'Finishing up!', {
		context: 'Shown during a jetpack authorization process, while we retrieve the info we need to show the last page'
	} );
}
```

Something just seems off to me about making this third condition more strict. Maybe if you explain in English the motivation I'll feel better about it.